crates/services/upgradable-executor/src/executor.rs

```diff
@@ -858,7 +922,10 @@ where
 
         let _ = preconfirmation_sender.send(preconfirmations).await;
 
-        Ok(result)
+        let mut g = storage_rec.lock();
```

Comment:
curious, if we `storage_rec` was an Option, and we performed a check like so -
```suggestion
        let storage_reads = match storage_rec {
            Some(storage_rec) => {
                let mut g = storage_rec.lock();
                core::mem::take(&mut *g)
            }
            None => vec![] /* replace with what you actually want / concrete type here */
        }
```

we would avoid the call to `.lock()` altogether. should describe the code better.

Reply:
I think the call to the `.lock()` is optimized away anyway. The compiler should see that this is the only reference to the mutex.